src/clover/fileparser.py

```diff
@@ -0,0 +1,2835 @@
+        except IndexError:
+            logger.error("Failed to determine buffer-water tank emission information.")
+            raise
+        except KeyError:
+            logger.error(
+                "Failed to determine buffer-water tank from the energy-system inputs "
+                "file."
+            )
+            raise
+        else:
+            logger.info("Buffer-water tank emission information successfully parsed.")
+
+    else:
+        buffer_tank_costs = None
+        buffer_tank_emissions = None
+        logger.info(
+            "Buffer-water tank disblaed in scenario file, skipping battery impact parsing."
+        )
+
+    if scenario.hot_water_scenario is not None:
+        logger.info("Parsing hot-water tank impact information.")
+        # Parse the hot-water tank costs information.
+        try:
+            hot_water_tank_costs: Optional[Dict[str, float]] = [
+                entry[COSTS]
+                for entry in tank_inputs
+                if entry[NAME]
+                == energy_system_inputs[ImpactingComponent.HOT_WATER_TANK.value]
+            ][0]
+        except IndexError:
+            logger.error("Failed to determine hot-water tank cost information.")
+            raise
+        except KeyError:
+            logger.error(
+                "Failed to determine hot-water tank from the energy-system inputs "
+                "file."
+            )
+            raise
+        else:
+            logger.info("Hot-water tank cost information successfully parsed.")
+
+        # Parse the hot-water tank emissions information.
+        try:
+            hot_water_tank_emissions: Optional[Dict[str, float]] = [
+                entry[EMISSIONS]
+                for entry in tank_inputs
+                if entry[NAME]
+                == energy_system_inputs[ImpactingComponent.HOT_WATER_TANK.value]
+            ][0]
+        except IndexError:
+            logger.error("Failed to determine hot-water tank emission information.")
+            raise
+        except KeyError:
+            logger.error(
+                "Failed to determine hot-water tank from the energy-system inputs "
+                "file."
+            )
+            raise
+        else:
+            logger.info("Hot-water tank emission information successfully parsed.")
+
+    else:
+        hot_water_tank_costs = None
+        hot_water_tank_emissions = None
+        logger.info(
+            "Hot-water tank disblaed in scenario file, skipping battery impact parsing."
+        )
+
+    return (
+        buffer_tank_costs,
+        buffer_tank_emissions,
+        clean_water_tank_costs,
+        clean_water_tank_emissions,
+        hot_water_tank_costs,
+        hot_water_tank_emissions,
+        tank_inputs,
+        tank_inputs_filepath,
+    )
+
+
+def _parse_minigrid_inputs(
+    convertors: Dict[str, Convertor],
+    debug: bool,
+    inputs_directory_relative_path: str,
+    logger: Logger,
+    scenario: Scenario,
+) -> Tuple[
```

Comment:
If I'm guessing what this does correctly: would the elements of this Tuple need to get updated if we added any extra elements to the inputs/outputs of the simulation(?)